src/node/node_state.h

```diff
@@ -1507,45 +1502,49 @@ namespace ccf
       // map the node id to a hostname and service and inform raft so that it
       // can add a new active configuration.
       network.nodes.set_local_hook(
-        [this](
-          kv::Version version, const Nodes::State& s, const Nodes::Write& w) {
+        [this](kv::Version version, const Nodes::Write& w) {
           auto configure = false;
-          std::unordered_set<NodeId> configuration;
+          std::unordered_set<NodeId> configuration =
+            consensus->get_latest_configuration();
 
           for (auto& [node_id, ni] : w)
           {
-            switch (ni.value.status)
+            if (kv::is_deleted(ni.version))
```

Comment:
Would nodes be ever deleted from the KV?

Reply:
I think not currently, but every hook should still be checking for deletions. Maybe this should be an exception instead?

I'd like to do a pass making sure all hooks handle deletions correctly, but I'll do this with #1208.